### PR TITLE
store: support serving downsampled data from bucket

### PR DESCRIPTION
**Pull request by fabxc**
@Bplotka 

Definitely some tests missing overall – will add some later.

### CODE REVIEW

File: pkg/store/bucket.go [hunk 1]
```diff
@@ -373,12 +386,15 @@ func (s *BucketStore) Info(context.Context, *storepb.InfoRequest) (*storepb.Info
...
+	chks []storepb.AggrChunk
 }
+
 type bucketSeriesSet struct {
-	set  []seriesEntry
-	i    int
-	chks []storepb.AggrChunk
+	set []seriesEntry
+	i   int
+	// chks []storepb.AggrChunk
```
Comment (Bplotka) [inline pkg/store/bucket.go]: do we need this comment?

File: pkg/store/bucket.go [hunk 2]
```diff
@@ -526,6 +540,54 @@ func (s *BucketStore) blockSeries(
...
+			if err != nil {
+				return errors.Errorf("aggregate %s does not exist", downsample.AggrCount)
+			}
+			out.Count = &storepb.Chunk{Type: storepb.Chunk_XOR, Data: x.Bytes()}
+		case storepb.Aggr_SUM:
+			x, err := ac.Get(downsample.AggrSum)
+			if err != nil {
+				return errors.Errorf("aggregate %s does not exist", downsample.AggrSum)
+			}
+			out.Count = &storepb.Chunk{Type: storepb.Chunk_XOR, Data: x.Bytes()}
```
Comment (Bplotka) [inline pkg/store/bucket.go]: `out.Count` ??
Not `out.Sum`?

File: pkg/store/bucket.go [hunk 3]
```diff
@@ -712,6 +775,131 @@ func (s *BucketStore) LabelValues(ctx context.Context, req *storepb.LabelValuesR
...
+		if v == x {
+			return i
+		}
+	}
+	return -1
+}
+
+// getFor returns a time-ordered list of blocks that cover date between mint and maxt.
+// Blocks with the lowest resolution possible but not lower than the given resolution are returned.
+func (s *bucketBlockSet) getFor(mint, maxt, minResolution int64) (bs []*bucketBlock) {
```
Comment (Bplotka) [inline pkg/store/bucket.go]: how we got from `req.MaxResolutionWindow` to this `minResolution` name here? Also
```
for ; i < len(s.resolutions) && s.resolutions[i] > minResolution; i++ {}
```
Min resolution make sense here, but we are putting max Resolution in this parameter... Not sure what's going on here ^^

Comment (fabxc) [reply pkg/store/bucket.go]: That's essentially the issue about intuitively of naming we ran into before. `MaxResolutionWindow` and `minResolution` are in fact the same thing. If we set an upper bound on how big the window we aggregate over is that implies a minimum resolution we want to guarantee. If we made the window bigger, resolution would decrease.

**Comment by fabxc** [comment]
Fixed and added some tests.

**Comment by fabxc** [comment]
Gave this another review myself. I'll merge this now. There's one remaining comment on the naming confusion. If we come up with a better variant we can always change this later.

**Comment by Bplotka** [comment]
Oke (:

**Comment by fabxc** [comment]
Is skiing weather so bad that you are on GH instead on 11am :P?

**Pull request merged**